content/docs/installation/best-practice.md

```diff
@@ -22,6 +22,120 @@ are designed for backwards compatibility rather than for best practice or maximu
+
+1. **TCP: cert-manager (controller) -> HashiCorp Vault (authentication and resource API endpoints)**:
+   The cert-manager controller may establish HTTPS connections to one or more Vault API endpoints,
+   if you are using the [Vault Issuer](../configuration/vault.md).
+   The target host and port of the Vault endpoints
+   are configured in the Issuer or ClusterIssuer resources.
+
+1. **TCP: cert-manager (controller) -> Venafi TLS Protect (authentication and resource API endpoints)**:
+   The cert-manager controller may establish HTTPS connections to one or more Venafi API endpoints,
+   if you are using the [Venafi Issuer](../configuration/venafi.md).
+   The target host and port of the Venafi API endpoints
+   are configured in the Issuer or ClusterIssuer resources.
+
+1. **TCP: cert-manager (controller) -> DNS API endpoints (for ACME DNS01)**:
+   The cert-manager controller may establish HTTPS connections to DNS API endpoints such as Amazon Route53,
+   and to any associated authentication endpoints,
+   if you are using the [ACME Issuer with DNS01 solvers](../configuration/acme/dns01/README.md#supported-dns01-providers).
+
+1. **UDP / TCP: cert-manager (controller) -> External DNS**:
+   If you use the ACME Issuer, the cert-manager controller may send
+   DNS queries to recursive DNS servers,
+   as part of the ACME challenge self-check process.
+   It does this to ensure that the DNS01 or HTTP01 challenge is resolvable,
+   before asking the ACME server to perform its checks.
+
+   In the case of DNS01 it may also perform a series of DNS queries to authoritative DNS servers,
+   to compute the DNS zone in which to add the DNS01 challenge record.
+   In the case of DNS01, cert-manager also [supports DNS over HTTPS](../releases/release-notes/release-notes-1.13.md#dns-over-https-doh-support).
+
+   You can choose the host and port of the DNS servers, using the following [controller flags](../cli/controller.md):
+   `--acme-http01-solver-nameservers`,
+   `--dns01-recursive-nameservers`, and
+   `--dns01-recursive-nameservers-only`.
+
+1. **TCP: ACME (Let's Encrypt) -> cert-manager (acmesolver)**:
+   If you use an ACME Issuer configured for HTTP01,
+   cert-manager will deploy an `acmesolver` Pod, a Service and an Ingress (or Gateway API) resource
+   in the namespace of the Issuer
+   or in the cert-manager namespace if it is a ClusterIssuer.
+   The ACME server will establish an HTTP connection to this Pod via your chosen ingress load balancer,
+   so your network policy must allow this.
+
+   > ℹ️ The acmesolver Pod **does not** require access to the Kubernetes API server.
+
+1. **TCP: Metrics Server -> cert-manager (controller)**:
+   The cert-manager controller has a metrics server which listens for HTTP connections on TCP port 9402.
+   Create a network policy which allows access to this service from your chosen metrics collector.
+
+### Calico Example
```

Comment:
Would it be better to use the standard `networkpolicies.networking.k8s.io/v1` as the example, rather than a provider specific version?

Or we could have 3 example implementations?:
- Standard `NetworkPolicy`
- Calico
- Cillium

Reply:
> Would it be better to use the standard `networkpolicies.networking.k8s.io/v1` as the example, rather than a provider specific version?

The standard Kubernetes NetworkPolicy doesn't allow you to refer to a `Service`, which is what I think is needed to reliably allow cert-manager Pods to connect to the Kubernetes API server, so I used Calico.

Alternative would be to document how to find the IP address(es) of your K8S API server replicas and hard code those in to a Kubernetes NetworkPolicy....but that seems likely to break if the addresses ever change.

Reply:
I was about to reply but I see what you mean, you would have to have a combination of:

```yaml
spec:
  egress:
    to:
      namespaceSelector: ..
      podSelector: ..
```

But looking in GKE, you can't see the kuber api-server pods to select?
Whereas in a `kind` cluster you can:

```shell
> k get po -n kube-system -l component=kube-apiserver
NAME                                       READY   STATUS    RESTARTS   AGE
kube-apiserver-argocd-demo-control-plane   1/1     Running   0          6h44m
```

I'll see if there is a "workaround" as this must be a fairly common problem, in that, I need to lock down access but enable app to connect to the k8s api.

Reply:
Probably no quick fix. It would seem the common answer is exactly what you said @wallrj.
Ref: https://stackoverflow.com/questions/58790124/whitelist-kube-apiserver-with-network-policy